content/posts/2025-07-28-Building_Your_First_A2A_Agent.adoc

```diff
@@ -0,0 +1,432 @@
+---
+layout: post
+title:  "Building your First A2A Agent"
+date:   2025-07-28
+tags:   generative-ai a2a agent wildfly
+author: ehsavoie
+description: How to create an A2A agent in WildFly using the AI Feature pack.
+---
+
+The https://a2aprotocol.ai/[A2A Protocol] is an open standard for agent interoperability, enabling seamless collaboration between AI agents across different platforms. This article will guide you through building your first A2A-enabled agent in WildFly. We'll leverage https://github.com/wildfly-extras/a2a-java-sdk-server-jakarta[the A2A Java SDK for Jakarta Servers] (which provides a JakartaEE integration of the https://github.com/a2aproject/a2a-java[A2A Java SDK]) and the https://github.com/wildfly-extras/wildfly-ai-feature-pack[WildFly AI Feature Pack] to create a powerful LLM-powered agent that uses a simple https://modelcontextprotocol.io/[Model Context Protocol] tool.
+
+== Prerequisites
+
+Before diving in, make sure you have:
+
+* JDK 17+
+* Apache Maven 3.8+
+* A Java IDE (e.g., Apache NetBeans, IntelliJ, VS Code)
+* A Gemini key
+* Python 3.10+
+* https://github.com/astral-sh/uv[uv] which is required to execute our MCP Tool written in Python.
+
+== Create the Weather Agent
```

Comment:
Just a nit, might be good to have a quick description of what you're going to create before this section.